frontend/src/use-cases/callback/index.js

```diff
@@ -1,16 +1,33 @@
-import React, { useEffect } from "react";
-import Cookies from "universal-cookie";
-const cookies = new Cookies();
+import React, { useEffect, useState, useContext } from "react";
+import { getToken, checkLogin } from "services/data.service";
+import { DigitLoading } from "@cthit/react-digit-components";
+import { Redirect } from "react-router-dom";
+import UserContext from "common/context/user-context";
 
 const Callback = ({ location }) => {
+    const [redirect, setRedirect] = useState(false);
+    const [, setUser] = useContext(UserContext);
+
     useEffect(() => {
         let params = new URLSearchParams(location.search);
-        cookies.set("AUTH_TOKEN", params.get("token"));
-        window.location.replace("/");
-        return () => {};
-    }, [location.search]);
+        getToken(params.get("code"))
+            .then(res => {
+                checkLogin(setUser);
+                setRedirect(true);
+            })
+            .catch(err => {
+                console.log(err);
+                console.log("Failed to fetch token");
+                setRedirect(true);
+            });
+    }, [location.search, setRedirect, setUser]);
 
-    return <div />;
+    return (
+        <>
+            <DigitLoading size={80} />
```

Comment:
Why make it larger? Is it really needed? I think it's nice to try to keep the size to the default one for all digIT websites